R/types.R

```diff
@@ -179,10 +179,8 @@ type_object <- function(
 convert_from_type <- function(x, type) {
   if (S7_inherits(type, TypeArray)) {
     if (S7_inherits(type@items, TypeBasic)) {
-      if (!type@items@required) {
-        is_null <- map_lgl(x, is.null)
-        x[is_null] <- list(NA)
-      }
+      is_null <- map_lgl(x, is.null)
+      x[is_null] <- list(NA)
```

Comment:
Just to confirm, this would convert `[1, 2, null]` from `list(1, 2, NULL)` to `c(1, 2, NA)`, but non-required tool arguments would still receive `NULL` if the model sends `"opt_arg": null`?

Reply:
None of the existing types correspond to `NULL` in my mind. Can you please give me a bit more detail?

Reply:
Sure, I'm mostly thinking of a tool with a non-required argument, e.g. something like this

```
btw_tool_session_package_info <- function(
  packages = "attached",
  dependencies = NULL
) {
  # ... get session info, dependencies optional
}
```

where `dependencies` has `required = FALSE` in the tool definition. I'm assuming that, as a non-required argument, the model might not include `dependencies` or it might send `{"packages": "attached", "dependencies": null}`. I think it'd be surprising to the tool author for the function to receive `dependencies = NA` instead of `dependencies = NULL`.